airflow/dag_processing/processor.py

```diff
@@ -533,6 +533,10 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
 
         # Add the errors of the processed files
         for filename, stacktrace in dagbag.import_errors.items():
+            dag_model = session.query(DagModel).filter(DagModel.fileloc == filename).first()
```

Comment:
I don't think this can be on just the first DagModel, don't we need to set it on all of them from that file?

Reply:
```suggestion
            session.query(DagModel).filter(DagModel.fileloc == filename).update({"has_import_errors": True})
```

might be better _I think_